entrypoint.sh

```diff
@@ -75,9 +82,9 @@ if grep -q "^--- .*d" "$onlineapp" > /dev/null 2>&1  ; then
         exec timeout -s KILL ${TIMEOUT:-30} bash -c "${DLANG_EXEC} -g $args "${d_files[@]:1}" $with_run "${d_files[0]}" ${run_args} | tail -n100000"
     fi
 elif  grep -qE "dub[.](sdl|json):" "$onlineapp" > /dev/null 2>&1  ; then
-    exec timeout -s KILL ${TIMEOUT:-30} dub -q --compiler=${DLANG_EXEC} --single --skip-registry=all "$onlineapp" ${run_args} | tail -n10000
+    exec timeout -s KILL ${TIMEOUT:-30} dub ${DUB_COMMAND} -q --compiler=${DLANG_EXEC} --single --skip-registry=all "$onlineapp" ${run_args} | tail -n10000
```

Comment:
I don't think `test` works with `--single`.
```console
% dub test --single source/scpp/build.d
Package cpp_build (configuration "application") defines no import paths, use {"importPaths": [...]} or the default package directory structure to fix this.
Generating test runner configuration 'cpp_build-test-application' for 'application' (executable).
Source file '/Users/geod24/projects/bpfk/agora/source/scpp/build.d' not found in any import path.
```

Script used is [here](https://github.com/bpfkorea/agora/blob/330cdaa4881ec7043249e862ffd262939c4d4802/source/scpp/build.d#L1-L7).

Reply:
Do you think that is a `dub` bug? It is listed as an option under the `test` [docs](https://dub.pm/commandline#test). 

I see some issues in the `dub` repo about single. It rebuilds every time, it doesn't work for dynamic libraries, and it can fail when the network is bad.

Reply:
@Geod24 After actually being able to run it myself, I can confirm that I run into that issue.

Using the most recent branch, I built with `docker build -t dlangtour/core-exec:dmd .` and ran 
```
source="/++dub.sdl: name\"foo\" \n dependency\"mir\" version=\"*\"+/ unittest { import mir.combinatorics, std.stdio; writeln([0, 1].permutations); } void main() {}"
bsource=$(echo -e "$source" | base64 -w0)
DOCKER_FLAGS="-unittest" docker run -e DOCKER_FLAGS --rm dlangtour/core-exec:dmd "$bsource"
```
and it printed
```
Package foo (configuration "application") defines no import paths, use {"importPaths": [...]} or the default package directory structure to fix this.
Source file '/sandbox/onlineapp.d' not found in any import path.
```
When excluding dependencies (code below), I get the same error. 
```
source="/++dub.sdl: name\"foo\" \n +/ unittest { assert(1 > 0); } void main() {}"
bsource=$(echo -e "$source" | base64 -w0)
DOCKER_FLAGS="-unittest" docker run -e DOCKER_FLAGS --rm dlangtour/core-exec:dmd "$bsource"
```
It's not as if you can remove `--single` when handling `dub test` only. That doesn't work. And you need `--single` when using `dub run`.